### PR TITLE
Prevent GraphQL query body to be huge and cause OOM

### PR DESCRIPTION
**Pull request by xgreenx**
Closes https://github.com/FuelLabs/fuel-core/issues/1971

## Checklist
- [x] New behavior is reflected in tests

### Before requesting review
- [x] I have reviewed the code myself

### CODE REVIEW

File: crates/fuel-core/src/service/config.rs [hunk 1]
```diff
@@ -129,6 +129,7 @@ impl Config {
                 max_queries_depth: 16,
                 max_queries_complexity: 20000,
                 max_queries_recursive_depth: 16,
+                max_queries_body_limit: 16 * 1024 * 1024,
```
Comment (Voxelot) [inline crates/fuel-core/src/service/config.rs:132]: Currently we support queries up to 7.5mb at the load balancer level. Should we match that here?

I think it's pretty common for RPC providers to support up to 10+ mb responses on other jsonrpc based chains.

cc @mchristopher

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - [#1992](https://github.com/FuelLabs/fuel-core/pull/1992): Parse multiple relayer contracts, `RELAYER-V2-LISTENING-CONTRACTS` env variable using a `,` delimiter.
 
 #### Breaking
+- [#2001](https://github.com/FuelLabs/fuel-core/pull/2001): Prevent GraphQL query body to be huge and cause OOM. The default body size is `256KB`. The limit can be changed by the `graphql-max-queries-body-limit` CLI argument.
```
Comment (Voxelot) [inline CHANGELOG.md:19]: Currently we support queries up to 7.5mb at the load balancer level. Should we match that with our default here?

I think it's pretty common for RPC providers to support up to 10+ mb responses on other jsonrpc based chains.

cc @mchristopher

Comment (xgreenx) [reply CHANGELOG.md:19]: This limit applies only to the query itself; it doesn't affect the response.

But I realized that we use a hex representation of transactions, so we need to support twice 256 KB. Maybe it is better to set it to 1 MB.

File: CHANGELOG.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - [#1980](https://github.com/FuelLabs/fuel-core/pull/1980): Add `Transaction` to relayer 's event filter
 
 #### Breaking
+- [#2001](https://github.com/FuelLabs/fuel-core/pull/2001): Prevent GraphQL query body to be huge and cause OOM. The default body size is `1MB`. The limit can be changed by the `graphql-max-queries-body-limit` CLI argument.
```
Comment (mchristopher) [inline CHANGELOG.md:21]: Can we reference `bytes` in this variable name? As I read it, it sounds like this value influences the max size of the graph query, not the total byte size of the request. Maybe something like `graphql-max-request-body-bytes`?

File: bin/fuel-core/src/cli/run/graphql.rs [hunk 4]
```diff
@@ -24,6 +24,14 @@ pub struct GraphQLArgs {
     #[clap(long = "graphql-max-recursive-depth", default_value = "16", env)]
     pub graphql_max_recursive_depth: usize,
 
+    /// The max body limit of the GraphQL query.
+    #[clap(
+        long = "graphql-max-queries-body-limit",
+        default_value = "1048576",
```
Comment (mchristopher) [inline bin/fuel-core/src/cli/run/graphql.rs:30]: As a default, this seems a bit excessive. The built-in default axum limit is 2MB, so I'm not sure that halving that is the best default. I'd think somewhere in the 256KB range as a default, given how much data traversal (i.e., computational expensive queries) can be easily put into a single request.

Comment (xgreenx) [reply bin/fuel-core/src/cli/run/graphql.rs:30]: The hex representation of the `Create` transaction can be `512KB`. I increased it to `1MB` just to allow additional data in the query or dry run two heavy transactions.

Before it was unlimited=)

Comment (mchristopher) [reply bin/fuel-core/src/cli/run/graphql.rs:30]: Got it, makes sense. I misread that we disabled the max limit previously - in that case, 1MB seems reasonable to me.

**Review by Voxelot — APPROVED** [review]

**Pull request merged**